torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3014,6 +3014,48 @@ def error_inputs_adaptive_max_pool3d(opinfo, device, **kwargs):
                      error_regex="Trying to create tensor with negative dimension")
 
 
+def sample_inputs_reduction_sparse(op_info, device, dtype, requires_grad, layout, blocksize=None, **kwargs):
+    if (((layout is torch.sparse_coo and not op_info.supports_sparse)
```

Comment:
Python nit:

You could use the [builtin all or any](https://docs.python.org/3/library/functions.html#any) functions and turn this into a list.

```
if any([layout is getattr(torch, 'sparse_' + layout_name) and not getattr(op_info, 'supports_sparse' + opinfo_name for (layout_name, opinfo_name) in zip(['coo', 'csr', 'csc', 'bsr', 'bsc'], ['', 'csr', 'csc', 'bsr', 'bsc'])):
    return
```

Not sure this is much better, just a suggestion.

Reply:
Now using:
```python
layout_name = str(layout).split('.', 1)[-1].rsplit('_coo', 1)[0]
op_supports_layout = getattr(op_info, 'supports_' + layout_name)
if not op_supports_layout:
    return
```